### DISCUSSION TITLE
git is not pushing a repo, wonder what is the mistake ? plz help

### DISCUSSION THREAD

**Discussion by Punith-sk** [community/community#132431] [Pull Requests]
### Select Topic Area

General

### Body

PC@******* MINGW64 ~/Desktop/alldesktop/ob/********* (main)
$ git push -u origin main
error: src refspec main does not match any
error: failed to push some refs to 'github.com-college:Punith-sk/crypto_levels.git'

**Comment by davevad93** [comment]
Hi @Punith-sk ,

This error usually occurs when no changes have been added from the working directory to the staging area or if you haven't created an initial commit, be sure to run the following commands in order:

```bash
git add .
```
```bash
git commit -m "Initial Commit"
```
```bash
git push origin main
```

**Comment by Punith-sk** [comment]
it turns out initiating the README was mandatory now it is working i mean the git was pushed. thank you for your reply

**Reply by davevad93** [reply]
You're welcome @Punith-sk , I'm glad you've solved it!
